src/main/java/io/divolte/server/IncomingRequestProcessor.java

```diff
@@ -65,42 +56,29 @@
     private final boolean keepDuplicates;
 
     public IncomingRequestProcessor(final ValidatedConfiguration vc,
-                                    @Nullable final KafkaFlushingPool kafkaFlushingPool,
-                                    @Nullable final HdfsFlushingPool hdfsFlushingPool,
-                                    @Nullable final LookupService geoipLookupService,
+                                    final String name,
+                                    final EventForwarder<AvroRecordBuffer> flushingPools,
+                                    final Optional<LookupService> geoipLookupService,
                                     final Schema schema,
                                     final IncomingRequestListener listener) {
-
-        this.kafkaFlushingPool = kafkaFlushingPool;
-        this.hdfsFlushingPool = hdfsFlushingPool;
+        this.flushingPools = flushingPools;
         this.listener = Objects.requireNonNull(listener);
 
-        keepCorrupted = !vc.configuration().incomingRequestProcessor.discardCorrupted;
+        final MappingConfiguration mappingConfiguration = vc.configuration().getMappingConfiguration(name);
+        keepCorrupted = !mappingConfiguration.discardCorrupted;
 
-        memory = new ShortTermDuplicateMemory(vc.configuration().incomingRequestProcessor.duplicateMemorySize);
-        keepDuplicates = !vc.configuration().incomingRequestProcessor.discardDuplicates;
+        memory = new ShortTermDuplicateMemory(vc.configuration().global.mapper.duplicateMemorySize);
```

Comment:
Is it possible to use something like vc.configuration().getGlobalMappingConfiguration().duplicateMemorySize or vc.globalConfiguration().duplicateMemorySize.. If the global.mapper namespace changes you only need 1 place to change it.